packages/amplify-graphql-transformer-core/src/utils/function-runtime.ts

```diff
@@ -0,0 +1,45 @@
+import { MappingTemplateProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { CfnFunctionConfiguration } from 'aws-cdk-lib/aws-appsync';
+import { GraphQLApi } from '../graphql-api';
+import { InlineTemplate } from '../cdk-compat';
+import { Construct } from 'constructs';
+
+type RuntimeSpecificFunctionProps = {
+  requestMappingTemplate?: string;
+  responseMappingTemplate?: string;
+  requestMappingTemplateS3Location?: string;
+  responseMappingTemplateS3Location?: string;
+  runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty;
+  code?: string;
+};
+
+export const getRuntimeSpecificFunctionProps = (
+  scope: Construct,
+  props: {
+    requestMappingTemplate: MappingTemplateProvider;
+    responseMappingTemplate: MappingTemplateProvider;
+    runtime?: CfnFunctionConfiguration.AppSyncRuntimeProperty;
+    api: GraphQLApi;
+  },
+): RuntimeSpecificFunctionProps => {
+  const { requestMappingTemplate, responseMappingTemplate, runtime, api } = props;
+
+  const requestTemplateLocation = requestMappingTemplate.bind(scope, api.assetProvider);
+  const responseTemplateLocation = responseMappingTemplate.bind(scope, api.assetProvider);
+
+  if (runtime?.name === 'APPSYNC_JS') {
```

Comment:
This gives me a bit of pause. I assume that your note about making minimal changes is why we've elected to special case the JS handling by assuming inline code, rather than allowing templates like the VTL template provider does? If so, I agree it's probably the right thing to do for this exercise.

Reply:
Your assumption is correct. I would've preferred to support `codeS3Location` as well here (i.e. do it right), but that got messy quickly because of VTL based assumptions throughout the codebase that templates are split into individual requests and responses -- something that is not the case with `code` (JS function resolvers).

Where both `responseMappingTemplateS3Location` and `requestMappingTemplateS3Location` are necessary for the VTL runtime, a single `codeS3Location` is required for JS runtime.

I'm hoping it's not as much work as it appeared, but I opted to ~kick the can down the road~ do this incrementally to reduce the risk of regressions in other transformers.

Reply:
> I opted to kick the can down the road ~~do this incrementally~~

Quality save. :)